src/test/java/org/wiremock/extensions/template/MockRequest.java

```diff
@@ -0,0 +1,384 @@
+package org.wiremock.extensions.template;
+
+import static com.github.tomakehurst.wiremock.common.Exceptions.throwUnchecked;
+import static com.github.tomakehurst.wiremock.common.ParameterUtils.getFirstNonNull;
+import static com.github.tomakehurst.wiremock.common.Strings.bytesFromString;
+import static com.github.tomakehurst.wiremock.http.HttpHeader.httpHeader;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.asList;
+
+import com.github.tomakehurst.wiremock.common.Urls;
+import com.github.tomakehurst.wiremock.http.*;
+import com.github.tomakehurst.wiremock.servlet.WireMockHttpServletMultipartAdapter;
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+import org.eclipse.jetty.server.MultiPartInputStreamParser;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MockRequest implements Request {
```

Comment:
Copied from [wiremock/wiremock](https://github.com/wiremock/wiremock). Any recommendations on making this class available to extensions?
- add it to `main` instead of `test` so it is part of WireMock
- add it to plugin so all extensions that use the plugin can use the class
- etc.

Reply:
Frankly I'm a little hesitant to add it. My idea was to have the template very minimalistic to just demonstrate the actual creation of extensions.

Maybe you can get in touch with the wiremock team to expose this (and potential other classes) as separate package to help writing extension tests.